veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,225 @@
+for them in v1.5.
+
+## Motivation
+
+In previous KubeVirt versions, a namespace admin could hotplug and unplug NICs.
+We want to ensure that this feature will keep working in v1.5 and future versions and enhance it so the migration will be
+automatically performed - without user intervention.
+
+## Goals
+
+- Integrate the NIC hotplug / hotunplug with the LiveUpdate rollout strategy for consistent user experience.
+- Enable namespace admins to perform NIC hotplug / unplug operations independently - without involving the cluster-admin 
+for each hotplug / hotunplug operation.
+
+## Non Goals
+
+## Definition of Users
+
+- Namespace administrators
+
+## User Stories
+
+As a namespace admin I would like to be able to:
+- Hotplug / unplug a secondary interface without involving the cluster admin.
+- Hotplug a new secondary network interface that uses bridge binding.
+- Hotunplug a secondary network interface that uses bridge binding.
+- Hotplug a new secondary network interface of type SR-IOV.
+- Use the staging rollout strategy so any hotplug / hotunplug operation will take effect only after restarting a VM.
+
+## Repos
+
+- kubevirt/kubevirt
+
+## Design
+
+### Integrate with LiveUpdate Rollout Strategy
+
+The CPU and memory hotplug features use the `LiveUpdate` [rollout strategy](https://kubevirt.io/user-guide/user_workloads/vm_rollout_strategies/)
+which automatically initiates a migration of the VM when certain spec fields are changed.
+
+Extend the feature to automatically initiate a live migration in case of NIC hotplug / hotunplug request.
+
+#### API Additions
+
+Currently, the CPU, memory and volume hotplug / hotunplug flows have a dedicated condition on the VMI object to mark the VMI
+for automatic migration:
+- HotVCPUChange
+- HotMemoryChange
+- VolumesChange
+
+Since we do not wish to introduce another resource-specific condition, a new generic allowed value will be added to `VirtualMachineInstanceConditionType`:
+
+```go
+VirtualMachineInstanceMigrationRequired VirtualMachineInstanceConditionType = "MigrationRequired"
+```
+
+Existing hotplug / hotunplug flows could be adjusted to use it.
+
+#### Migration Based Hotplug Flow
+
+1. The VM owner adds one or more new secondary vNICs and matching networks to a running VM.
+2. The VM controller patches the VMI object with the additional relevant vNICs and networks.
+3. The VMI controller:
+   
+   - Patches the virt-launcher pod's Multus network attachment annotation (`k8s.v1.cni.cncf.io/networks`) - since the dynamic-networks-controller is not present, this will not have an affect.
+   - Updates the VMI interface status.
+   - Adds the `MigrationRequired` condition to the VMI with its status set to `False` and a timestamp.
+4. On a future iteration of the reconciliation loop, the VMI controller will update the `MigrationRequired` condition status to `True` (Following no change in the interface statuses for more than TBD seconds after the initial timestamp set on the condition).
```

Comment:
The information in the brackets is important, perhaps add it as a bullet.
As this is a formal design proposal, we better provide an initial estimation to the backoff. A range may be provided and experimented on with different clusters.

Reply:
> The information in the brackets is important, perhaps add it as a bullet.

The wording was adjusted.

> As this is a formal design proposal, we better provide an initial estimation to the backoff. A range may be provided and experimented on with different clusters.

Currently the text contains "ten seconds".
Lets not set it in stone.